source/view.js

```diff
@@ -3162,7 +3165,16 @@ view.TensorSidebar = class extends view.ObjectSidebar {
         if (value.type) {
             const item = new view.ValueView(this._view, value, '');
             this.add('type', item);
-            item.toggle();
+        }
+
+        if (value.initializer && value.initializer.category === 'Initializer') {
```

Comment:
Is the category check used to gate against empty tensors?

Reply:
It's used to filter out some of the tensors for which metrics are not relevant/ not applicable. For example, `onnx::Unsqueeze_840` specifies the dimension of unsqueezing, metrics are not applicable here.

Reply:
As written, empty initializer tensors are still considered worthy of metrics, although maybe they shouldn't be.

Reply:
Got it. `Initializer` is not used consistently, across formats and operators these types of tensor cannot be determined generically. Another approach could be filtering vectors with small length. Is that worth special casing or up to the user to interpret the result.

Reply:
> Initializer is not used consistently, across formats and operators these types of tensor cannot be determined generically

That makes sense.

A lot of the logic around "should metrics be applied to this tensor/node" must be done on a per-format basis, since you need the context of the operation to determine whether metrics are relevant. For example, it's difficult to tell whether sparsity applies to a tensor like `onnx::Unsqueeze_840` without knowing the context of ONNX and the unsqueeze operator.

As I see it, we have two design philosophies: In the case that the format does not provide a format-specific overload, then we can either *show default metrics* or *do not show default metrics*.

If we want to show default (sometimes unapplicable) metrics, we should just show them for all non-empty tensors and the user will know that formats with default metrics are not always reliable.
```javascript
if (!tensor.empty) {
```

If we don't want to show default metrics, then there's no point in implementing metrics on the `view.js` level. We should only be relying on format-specific implementations to provide metrics.